armi/physics/fuelCycle/fuelHandlers.py

```diff
@@ -1460,29 +1461,78 @@ def _swapFluxParam(self, incoming, outgoing):
+            mgFluxOutNew = resampleStepwise(meshIn, mgFluxIn, meshOut)
+            powerOutNew = resampleStepwise(meshIn, powerIn, meshOut)
+            fluxInNew = resampleStepwise(meshOut, fluxOut, meshIn)
+            mgFluxInNew = resampleStepwise(meshOut, mgFluxOut, meshIn)
+            powerInNew = resampleStepwise(meshOut, powerOut, meshIn)
+
+            # load the new outgoing values into place
+            for b, flux, mgFlux, power in zip(
+                outgoing, fluxOutNew, mgFluxOutNew, powerOutNew
+            ):
+                b.p.flux = flux
+                b.p.mgFlux = mgFlux
+                b.p.power = power
+                b.p.pdens = power / b.getVolume()
+
+            # load the new incoming values into place
+            for b, flux, mgFlux, power in zip(
+                incoming, fluxInNew, mgFluxInNew, powerInNew
+            ):
+                b.p.flux = flux
+                b.p.mgFlux = mgFlux
+                b.p.power = power
+                b.p.pdens = power / b.getVolume()
+
+            return
+
+        # Since the axial mesh points match, do the simple swap
         for bi, (bIncoming, bOutgoing) in enumerate(zip(incoming, outgoing)):
-            if (
-                bi not in self.cs["stationaryBlocks"]
-            ):  # stationary blocks are already swapped.
-                incomingFlux = bIncoming.p.flux
-                incomingMgFlux = bIncoming.p.mgFlux
-                incomingPower = bIncoming.p.power
-                outgoingFlux = bOutgoing.p.flux
-                outgoingMgFlux = bOutgoing.p.mgFlux
-                outgoingPower = bOutgoing.p.power
-                if outgoingFlux > 0.0:
-                    bIncoming.p.flux = outgoingFlux
-                    bIncoming.p.mgFlux = outgoingMgFlux
-                    bIncoming.p.power = outgoingPower
-                if incomingFlux > 0.0:
-                    bOutgoing.p.flux = incomingFlux
-                    bOutgoing.p.mgFlux = incomingMgFlux
-                    bOutgoing.p.power = incomingPower
+            if bi in self.cs["stationaryBlocks"]:
+                # stationary blocks are already swapped
+                continue
+
+            incomingFlux = bIncoming.p.flux
+            incomingMgFlux = bIncoming.p.mgFlux
+            incomingPower = bIncoming.p.power
+            outgoingFlux = bOutgoing.p.flux
+            outgoingMgFlux = bOutgoing.p.mgFlux
+            outgoingPower = bOutgoing.p.power
+
+            if outgoingFlux > 0.0:
+                bIncoming.p.flux = outgoingFlux
+                bIncoming.p.mgFlux = outgoingMgFlux
+                bIncoming.p.power = outgoingPower
+                bIncoming.p.pdens = outgoingPower / bIncoming.getVolume()
+
+            if incomingFlux > 0.0:
+                bOutgoing.p.flux = incomingFlux
+                bOutgoing.p.mgFlux = incomingMgFlux
+                bOutgoing.p.power = incomingPower
+                bOutgoing.p.pdens = incomingPower / bOutgoing.getVolume()
```

Comment:
This is not specifically related to your PR, but I am confused about this hunk of code. I am guessing this is meant to catch non-physical flux values, but such a consideration seems like it should be handled more aggressively. 

I imagine that the conditionals enforced here could lead to weird issues with. But also, the code does not complain if the swap doesn't take place, which I find to be a bad idea.

Do you have any thoughts on this? Maybe I am misunderstanding its purpose.

Reply:
To be 100% honest, I don't understand why this is in place.

It could use a comment, to explain, maybe.  @ntouran  ?

As a base line, I suppose I just tried to change the code as little as possible.